lobby/src/main/java/org/triplea/lobby/server/controller/rest/ModeratorActionController.java

```diff
@@ -0,0 +1,84 @@
+package org.triplea.lobby.server.controller.rest;
+
+import static spark.Spark.before;
+import static spark.Spark.halt;
+import static spark.Spark.path;
+import static spark.Spark.post;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.triplea.lobby.server.db.Database;
+import org.triplea.lobby.server.db.HashedPassword;
+import org.triplea.lobby.server.db.UserController;
+
+import com.google.common.base.Splitter;
+
+import games.strategy.engine.lobby.server.userDB.DBUser;
+import spark.Request;
+import spark.Response;
+
+class ModeratorActionController {
+  private final UserController controller;
+
+  ModeratorActionController(final Database database) {
+    this.controller = new UserController(database);
+  }
+
+
+  void initializeRoutes() {
+    path("/moderate", () -> {
+      before("/*", this::authorize);
+      post("/:username/set-permission/:rank", this::setRole);
+    });
+  }
+
+  static Map<String, String> parseBodyToMap(final String body) {
+    return Splitter.on('&').splitToList(body)
+        .stream()
+        .map(Splitter.on('=')::splitToList)
+        .collect(Collectors.toMap(list -> list.get(0), list -> list.get(1)));
+  }
+
+  void authorize(final Request req, final Response res) {
```

Comment:
DropWizard is still a contender for HTTP server. Let's try to write our backend logic so we can swap out the HTTP server without too much issue.